### PR TITLE
Some refactorings in addition to #3338

### PR DESCRIPTION
**Pull request by MaxGraey**
See discussion in https://github.com/WebAssembly/binaryen/pull/3303#issuecomment-725632036
+ some refactorings

### CODE REVIEW

File: src/passes/OptimizeInstructions.cpp [hunk 1]
```diff
@@ -1976,6 +1977,12 @@ struct OptimizeInstructions
              op == Abstract::getBinary(type, Abstract::LtS) ||
              op == Abstract::getBinary(type, Abstract::GeS) ||
              op == Abstract::getBinary(type, Abstract::GtS))) {
+          // if (auto* c = inner->right->dynCast<Const>()) {
+          //   // skip signed(x - min_s) <=> 0 case
+          //   if (c->value.isSignedMin()) {
+          //     return nullptr;
+          //   }
+          // }
```
Comment (kripken) [inline src/passes/OptimizeInstructions.cpp:1985]: I don't think this is the right approach. This only works if we know the constant at compile time. But we need to handle the case of a value only known at runtime.

Let me open a PR with what I think we have to do.

Comment (MaxGraey) [reply src/passes/OptimizeInstructions.cpp]: Yes, that's already outdated approach. With recent commit I removed simplification for `x - y <> 0` cases. And only leave for `x - y == 0` and `x - y != 0`

**Comment by MaxGraey** [comment]
@kripken could you take a look?

**Review by kripken — APPROVED** [review]
Has this been fuzzed?

**Comment by MaxGraey** [comment]
> Has this been fuzzed?

No, but it's nothing changed. Just simplify syntax via use namespace)

**Comment by kripken** [comment]
There's always a chance of bugs :wink: But, fair point, the risk is pretty small here, and we see no tests change.

**Pull request merged**